test/common/testlib.py

```diff
@@ -1305,6 +1307,23 @@ def write_file(self, path, content):
         if self.is_nondestructive():
             self.addCleanup(m.execute, "rm -f {0}".format(path))
 
+    def check_coverage(self):
+        """Collect code coverage result and save to file
+
+        Hash result as part of file name, like "coverage-<coverage hash result>.json",
+        to avoid duplicate coverage file and save file to ".nyc_output" folder.
+        """
+        coverage_data = self.browser.eval_js("window.__coverage__", no_trace=True)
+        str_coverage = json.dumps(coverage_data)
+        hash_str = hashlib.sha256(str_coverage.encode('utf-8')).hexdigest()
+
+        cov_out_dir = pathlib.Path(".nyc_output")
+        cov_out_dir.mkdir(exist_ok=True)
+
+        cov_out_file = cov_out_dir / "coverage-{}.json".format(hash_str)
+        # avoid duplecate result file
+        if not cov_out_file.is_file():
+            cov_out_file.write_text(str_coverage)
```

Comment:
we collect axe report in very similar fashion (check `check_axe` in the same file) and we do:
```
        filename = "{0}{1}-axe.json.gz".format(label or self.label(), suffix)   
        with gzip.open(filename, "wb") as f:                                    
            f.write(json.dumps(report).encode('UTF-8'))                         
        print("Wrote accessibility report to " + filename)                      
        attach(filename)  
```
and I think you would need to do something similar (like gzip, print that there was something saved and `attach` it, so it is actually available in CI)

Reply:
It's not a report, it's used by the code [here](https://github.com/weldr/cockpit-composer/blob/7f295a8ed5da2dd91f3f0328f480dc6beeb3e72c/test/check-application#L166). The report can be viewed from [codecov.io](https://codecov.io/gh/weldr/cockpit-composer). So we do not need these coverage files locally.